script/LaunchProjectAndDeployDomJuice.s.sol

```diff
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import {Script} from "forge-std/Script.sol";
+import {DominantJuice} from "../src/DominantJuice.sol";
+import {HelperConfig} from "./HelperConfig.s.sol";
+import {IJBController3_1} from "@jbx-protocol/juice-contracts-v3/contracts/interfaces/IJBController3_1.sol";
+import {IJBFundingCycleBallot} from "@jbx-protocol/juice-contracts-v3/contracts/interfaces/IJBFundingCycleBallot.sol";
+// import {IJBOperatorStore} from "@jbx-protocol/juice-contracts-v3/contracts/interfaces/IJBOperatorStore.sol";
+import {IJBProjects} from "@jbx-protocol/juice-contracts-v3/contracts/interfaces/IJBProjects.sol";
+import {IJBSingleTokenPaymentTerminalStore3_1_1} from
+    "@jbx-protocol/juice-contracts-v3/contracts/interfaces/IJBSingleTokenPaymentTerminalStore3_1_1.sol";
+import {IJBPaymentTerminal} from "@jbx-protocol/juice-contracts-v3/contracts/interfaces/IJBPaymentTerminal.sol";
+import {JBETHPaymentTerminal3_1_1} from "@jbx-protocol/juice-contracts-v3/contracts/JBETHPaymentTerminal3_1_1.sol";
+import {JBFundAccessConstraints} from "@jbx-protocol/juice-contracts-v3/contracts/structs/JBFundAccessConstraints.sol";
+import {JBFundingCycleData} from "@jbx-protocol/juice-contracts-v3/contracts/structs/JBFundingCycleData.sol";
+import {JBFundingCycleMetadata} from "@jbx-protocol/juice-contracts-v3/contracts/structs/JBFundingCycleMetadata.sol";
+import {JBGlobalFundingCycleMetadata} from
+    "@jbx-protocol/juice-contracts-v3/contracts/structs/JBGlobalFundingCycleMetadata.sol";
+import {JBGroupedSplits} from "@jbx-protocol/juice-contracts-v3/contracts/structs/JBGroupedSplits.sol";
+//import {JBOperatable} from "@jbx-protocol/juice-contracts-v3/contracts/abstract/JBOperatable.sol";
+//import {JBOperations} from "@jbx-protocol/juice-contracts-v3/contracts/libraries/JBOperations.sol";
+//import {JBOperatorData} from "@jbx-protocol/juice-contracts-v3/contracts/structs/JBOperatorData.sol";
+import {JBProjectMetadata} from "@jbx-protocol/juice-contracts-v3/contracts/structs/JBProjectMetadata.sol";
+//import {LaunchProjectData} from "./structs/LaunchProjectData.sol";
+
+contract LaunchAndDeploy is Script {
+    // Juicebox contracts
+    struct Contracts {
+        IJBController3_1 controller; // Controller that configures funding cycles
+        IJBSingleTokenPaymentTerminalStore3_1_1 paymentTerminalStore; // Stores all payment terminals
+        JBETHPaymentTerminal3_1_1 ethPaymentTerminal; // Default ETH payment terminal
+    }
+
+    Contracts contracts;
```

Comment:
using these fields for the script are really globals, which means that the two functions are interrelated in ways that aren't obvious. In other words, you can't call `launchProjectFor` without calling `run` because these variables haven't been set up. Scripts really shouldn't have any fields at all. They should pass all variables to the next function, or let the function go get it themselves if needed.

Reply:
Seems I need to brush up on scripting or tutorials have led me astray. Might need a call for this, but let’s save this until we are at the testnet stage, if that works for you.